mne/io/edf/edf.py

```diff
@@ -114,6 +114,11 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         annotmap = self._raw_extras[fi]['annotmap']
         subtype = self._raw_extras[fi]['subtype']
 
+        # Interpolating stim chan except it later gets overwritten.
+        if ((not (annot and annotmap or tal_channel is not None) and
+                (stim_channel in sel))):
+            warn('Interpolating stim channel. Events may jitter.')
```

Comment:
`sel` just determines what channels are being read at the moment.

Look at line 368 in _get_edf_info, `tal_channel` is in there. I can take over from this commit if it's too annoying to sort out. It is a bit complicated.